app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -918,3 +918,27 @@ export const appendSignedMessage = async (ethereumClientService: EthereumClientS
 		signedMessages: simulationState.signedMessages.concat(signedMessage)
 	}
 }
+
+// takes the most recent block that the application is querying and does the calculation based on that
+export const getSimulatedFeeHistory = async (ethereumClientService: EthereumClientService, simulationState: SimulationState | undefined, request: FeeHistory): Promise<EthGetFeeHistoryResponse> => {
+	//const numberOfBlocks = request.params[0]
+	const blockTag = request.params[1]
+	const rewardPercentiles = request.params[2]
+	const newestBlock = await getSimulatedBlock(ethereumClientService, simulationState, blockTag, true)
+	const blocksToConsider = [newestBlock]
+	const oldestBlock = blocksToConsider[0]?.number
+	if (oldestBlock === undefined) throw new Error('oldest block was undefined')
+	return {
+		baseFeePerGas: blocksToConsider.map((block) => block.baseFeePerGas).filter((baseFeePerGas): baseFeePerGas is bigint => !!baseFeePerGas),
+		gasUsedRatio: blocksToConsider.map((block) => Number(block.gasUsed) / Number(block.gasLimit)).filter((gasUsedRatio): gasUsedRatio is number => !!gasUsedRatio),
```

Comment:
These maps seem unnecessary when the `blocksToConsider` is always a single item tuple.  Why not just do something like
```suggestion
		baseFeePerGas: !!(newestBlock.baseFeePerGas) ? [newestBlock.baseFeePerGas] : [],
		gasUsedRatio: !!(newestBlock.gasUsed) ? [Number(newestBlock.gasUsed) / Number(block.gasLimit)] : [],
```

Reply:
Same question for the more complicated `rewardPercentiles`.